src/server.js

```diff
@@ -44,7 +45,7 @@ export default function hypernova(userConfig, onServer) {
 
   logger.init(config.logger, config.loggerInstance);
 
-  const app = createApplication();
+  const app = config.expressInstance || createApplication();
```

Comment:
instead of passing an instance, it would probably be better to pass a `createApplication` function.

We’d also want to throw if that function failed to return an express instance.